### PR TITLE
[HttpKernel] Remove BC layer in DebugHandlersListener

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       |
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by lyrixx** [comment]
Oups, looks like I never open the PR for the first commit. Will do that asap

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php [hunk 1]
```diff
@@ -25,7 +25,6 @@
                 null, // Log levels map for enabled error levels
                 param('debug.error_handler.throw_at'),
                 param('kernel.debug'),
-                service('debug.file_link_formatter'),
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php]: This means that we just broke cross-compat between 5.4 & 6.0. composer.json needs to be updated.
On 5.4, what about still allowing v6 by doing a reflection check?

**Comment by nicolas-grekas** [comment]
Please rebase to restart tests :pray:

**Comment by nicolas-grekas** [comment]
We are missing a BC layer in 5.4 first to deprecate the argument and accept the new signature.
Could you please have a look?

**Comment by lyrixx** [comment]
I was on holiday, I'll have a look on Friday 👍🏼 
About the BC: The class is internal now. see https://github.com/symfony/symfony/pull/39016#discussion_r518842159

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.php]: see https://github.com/symfony/symfony/pull/42235

**Comment by lyrixx** [comment]
I updated the PR with the new approach

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Pull request merged**